quill-sql/src/main/scala/io/getquill/context/sql/idiom/SqlIdiom.scala

```diff
@@ -273,24 +274,25 @@ trait SqlIdiom extends Idiom {
   }
 
   implicit def propertyTokenizer(implicit astTokenizer: Tokenizer[Ast], strategy: NamingStrategy): Tokenizer[Property] = {
-    def unnest(ast: Ast): (Ast, String) =
+
+    def unnest(ast: Ast): (Ast, List[String]) =
       ast match {
         case Property(a, name) if (name.matches("_[0-9]*")) =>
           unnest(a) match {
             case (ast, nestedName) =>
-              (ast, s"$nestedName$name")
+              (ast, nestedName :+ name)
```

Comment:
what do you think about replacing all Lists with Vectors for better append performance? (O(n) -> O(1))

Reply:
Good question. I'm planning to review the overall performance of query compilation. I'll profile and see if it'd be a good optimization before releasing 3.0